packages/react-instantsearch-hooks-dom/src/ui/SearchBox.tsx

```diff
@@ -2,26 +2,35 @@ import React from 'react';
 
 import { cx } from './lib/cx';
 
-import type {
-  ChangeEvent,
-  FormEvent,
-  RefObject,
-  JSXElementConstructor,
-} from 'react';
-
-export type SearchBoxProps = React.HTMLAttributes<HTMLDivElement> & {
-  inputRef: RefObject<HTMLInputElement>;
-  isSearchStalled: boolean;
-  onChange(event: ChangeEvent): void;
-  onReset(event: FormEvent): void;
-  onSubmit?(event: FormEvent): void;
-  placeholder?: string;
-  value: string;
-  resetIconComponent?: JSXElementConstructor<Record<string, any>>;
-  submitIconComponent?: JSXElementConstructor<Record<string, any>>;
-  loadingIconComponent?: JSXElementConstructor<Record<string, any>>;
+export type SearchBoxTranslations = {
+  /**
+   * The alternative text of the submit button.
+   */
+  submitTitle: string;
+  /**
+   * The alternative text of the reset button.
+   */
+  resetTitle: string;
 };
 
+export type SearchBoxProps = Omit<
+  React.HTMLAttributes<HTMLDivElement>,
+  'onSubmit' | 'onReset'
+> &
+  Pick<React.HTMLAttributes<HTMLFormElement>, 'onSubmit' | 'onReset'> &
+  Pick<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    'placeholder' | 'onChange'
+  > & {
+    value: string;
```

Comment:
I didn't pick it from `React.InputHTMLAttributes<HTMLInputElement>` because the types are too wide (`string | number | string[] | undefined`) and we're always passing a string anyway, so this avoids extra and useless runtime code.